Findbar fixes

Quicksaver opened the pull request:
Fixes #401:
- Don't find immediately on every key pressed while typing complex words (IME compositing).
- Don't show the findbar when searching through the command line and there are no matches.

Quicksaver commented:
Ah hell that merge commit wasn't supposed to be in here... I'll try again.

Quicksaver commented:
Hmm the pull itself doesn't actually have any other changes because of that, so it should still pull only the changes from the other commits. I'm leaving as is for now, but do let me know if you'd like me to create a cleaner pull (git is tricky sometimes...)

gkatsev commented:
Thanks, I'll take a look tomorrow. Didn't get a chance to review it today.

gkatsev commented on common/content/commandline.js:
```diff
@@ -421,6 +421,16 @@ const CommandLine = Module("commandline", {
         }
     },
 
+    _isIMEComposing: false,
```
maybe this should be called something slightly more general?
While it does mostly goes to fix IME input, it could be a useful property to have and I don't think it is actually set when IME is active.

gkatsev replied:
or is `compositionstart` and `compositionend` specific events for IME?

gkatsev replied:
Ah, sorry, yes, this is IME specific.

Quicksaver replied:
Yep. :)

gkatsev commented on common/content/finder.js:
```diff
@@ -124,15 +128,27 @@ var Finder = Module("finder", {
+            // even though that way we lose any kind of no matches notification until we
+            // stop typing altogether; something to work on at a later time:
+            // - show the quick findbar which will hide after a few seconds?
+            // - or notify the user somehow in the command line itself?
```
yep, we should have the commandline turn red or something to signal this. We can either work on it as part of this PR or we can merge this one in and then work on it in another one.

Quicksaver replied:
I didn't do this in this patch at least because I didn't know how to tackle it. The command line's echo methods only really show stuff when not inputting. I didn't want to create new methods/statuses/whatever because I don't really know what I'm doing there. :stuck_out_tongue_winking_eye:

gkatsev commented on common/content/finder.js:
```diff
@@ -124,15 +128,27 @@ var Finder = Module("finder", {
             // PDF.JS files are different, they use their own messages to communicate the results.
             // So we piggyback the end changes to the findbar when there are any.
-            findbar._original_updateControlState = findbar.updateControlState;
+            findbar._vimpbackup_updateControlState = findbar.updateControlState;
```
Is this rename so that we're unlikely to tread on someone else's steps if they were doing a similar thing in another addon?

Quicksaver replied:
Yeah, it was just a thought that occurred to me. I have seen add-ons backup ```.method``` into the same object as ```._method```, and I've also seen variants of ```.original_method``` and ```.backup_method``` as well. I thought specificity to Vimperator wouldn't hurt.

gkatsev commented on common/content/finder.js:
```diff
@@ -274,6 +290,18 @@ var Finder = Module("finder", {
+        if (config.name == 'Vimperator') {
+            // Same thing as the note above for not showing the findbar on failed results
+            // through the command line, but for compatibility with FindBar Tweak.
+            events.addSessionListener(window, "WillOpenFindBar", function (event) {
```
will findbar tweak respect the options above? I would prefer to keep other addon-specific fixed out of the codebase if at all possible.

Quicksaver replied:
FBT expects that when the original ```open``` method always leaves the findbar unhidden, which it won't because of the change above.

But I see your point, and thinking about it further it's probably better that I just add checks in FBT to see if the findbar actually opened or not after that was called. I'll upload a commit later today removing this part.

gkatsev commented:
Thanks! made some comments.

Quicksaver commented:
Expect a new commit later today to remove that FBT bit as soon as I fix it on my side as well. BTW do you think just turning the commandline red on failed matches will be enough? It could reuse the style for when it shows errors and stuff, maybe.

gkatsev commented:
Not positive exactly how it should look like. I'll think about it today.

Quicksaver commented:
Two new commits:
- removed the FBT bits as agreed,
- addressed https://github.com/vimperator/vimperator-labs/issues/401#issuecomment-180522987 which did make for a poor UX.

gkatsev commented:
LGTM. Thanks.
